features/exploredash/src/main/java/org/dash/wallet/features/exploredash/ui/SearchFragment.kt

```diff
@@ -175,6 +183,34 @@ class SearchFragment : Fragment(R.layout.fragment_search) {
         viewModel.allMerchantLocations.observe(viewLifecycleOwner) { merchantLocations ->
             merchantLocationsAdapter.submitList(merchantLocations)
         }
+
+        syncProgress.observe(viewLifecycleOwner) { syncProgress ->
+            if (syncProgress != null) {
+                when (syncProgress.status) {
+                    Status.LOADING -> {
+                        binding.apply {
+                            syncStatus.isVisible = true
+                            progress.isVisible = true
+                            progress.progress = syncProgress.data?.toInt() ?: 0
+                            syncMessage.text = getString(R.string.sync_in_progress_not_complete)
+                        }
+                    }
+                    Status.SUCCESS -> binding.syncStatus.isVisible = false
+                    Status.ERROR -> {
+                        binding.apply {
+                            syncStatus.isVisible = true
+                            syncStatus.setBackgroundResource(R.color.red_300)
+                            progress.isVisible = false
+                            when (syncProgress.exception) {
+                                is FileNotFoundException -> syncMessage.text = getString(R.string.sync_in_progress_file_not_found)
+                                else -> syncMessage.text = getString(R.string.sync_in_progress_error)
+                            }
+                        }
+                    }
+                    Status.CANCELED -> binding.syncStatus.isVisible = true
```

Comment:
Should sync status be visible in this case and if yes, don't we need to set some message?

Reply:
Yes, I added some text for this case also, though, currently there is nothing that triggers a canceled state.